wkcuber/wkcuber/export_wkw_as_nifti.py

```diff
@@ -0,0 +1,206 @@
+    parser = ArgumentParser()
+
+    parser.add_argument(
+        "--source_path", "-s", help="Directory containing the wkw file.", required=True
+    )
+
+    parser.add_argument(
+        "--destination_path",
+        "-d",
+        help="Output directory for the generated nifti files.",
+        required=True,
+    )
+
+    parser.add_argument("--name", "-n", help="Name of the nifti", default="")
+
+    parser.add_argument(
+        "--bbox",
+        help="The bounding box of which the nifti file should be generated."
+        "The input format is x,y,z,width,height,depth."
+        "(By default, data for the full bounding box of the dataset is generated)",
+        default=None,
+        type=parse_bounding_box,
+    )
+
+    parser.add_argument(
+        "--mag", "-m", help="The magnification that should be read", default=1
+    )
+
+    parser.add_argument(
+        "--downsample", help="Downsample each nifti image", default=1, type=int
+    )
+
+    parser.add_argument(
+        "--original_bbox",
+        help="Easy way to add padding to the generated file. Format is: x,y,z,width,height,depth"
+        "x,y,z is the offset of the wkw layer into the final bounding box; "
+        "width,height,depth corresponds to final dimensions",
+        default=None,
+        type=parse_bounding_box,
+    )
+
+    parser.add_argument(
+        "--bounding_box_crop",
+        help="Easy way to crop the final file. Is applied AFTER everything else (including padding)."
+        "Format is: x,y,z,width,height,depth",
+        default=None,
+        type=parse_bounding_box,
+    )
+
+    add_verbose_flag(parser)
+    add_distribution_flags(parser)
+
+    return parser
+
+
+def export_layer_to_nifti(
+    wkw_file_path: Path,
+    bbox: Dict,
+    mag: Mag,
+    layer_name: str,
+    destination_path: Path,
+    name: str,
+    original_bbox_size: Optional[List[int]] = None,
+    offset_into_orginal_bbox: Optional[List[int]] = None,
+    bounding_box_crop: Optional[List[int]] = None,
+) -> None:
+    wk_ds = Dataset(wkw_file_path)
+    layer = wk_ds.get_layer(layer_name)
+    mag_layer = layer.get_mag(mag)
+
+    is_segmentation_layer = layer.category == LayerCategories.SEGMENTATION_TYPE
+
+    data = mag_layer.read(bbox["topleft"], bbox["size"])
+    data = data.transpose(1, 2, 3, 0)
+    logging.info(f"Shape with layer {data.shape}")
+
+    data = np.array(data)
+    if is_segmentation_layer and data.max() > 0:
+        factor = np.iinfo("uint8").max / data.max()
+        data = data * factor
+        data = data.astype(np.dtype("uint8"))
+
+    if original_bbox_size and offset_into_orginal_bbox:
+        padding_per_axis = []
+
+        for i in range(3):
+            if original_bbox_size[i] == data.shape[i]:
+                padding_per_axis.append((0, 0))
+            else:
+                left_pad = offset_into_orginal_bbox[i]
+                right_pad = (
+                    original_bbox_size[i] - offset_into_orginal_bbox[i] - data.shape[i]
```

Comment:
the containment relationships I added to the argparse help should also be validated here. you can use `BoundingBox.contains_bbox` for that